src/controllers/controller.bar.js

```diff
@@ -496,6 +464,42 @@ export default class BarController extends DatasetController {
 
 }
 
+BarController.id = 'bar';
```

Comment:
I think based off @sgratzl 's findings this will stop the controller from being shaken out in WebPack and maybe we need to wrap in a pure annotated iife?

Reply:
Not sure how to get rollup to preserve those annotations.

Reply:
https://github.com/terser/terser/issues/724

Reply:
Looks like later rollup shakes the `prototype` variant too:
[repl](https://rollupjs.org/repl/?version=2.15.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QnRlc3QlN0QlMjBmcm9tJTIwJy4lMkZtb2RlbCclM0IlNUNuJTVDbnRlc3QoKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1vZGVsLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNsYXNzJTIwVGVzdCUyMCU3QiU1Q24lMjAlMjAlMjAlMjB0ZXN0KCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwY29uc29sZS5sb2coJ3Rlc3QnKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlNUNuJTdEJTVDbiU1Q25UZXN0LmlkJTIwJTNEJTIwJ0EnJTNCJTVDbiU1Q25leHBvcnQlMjBjbGFzcyUyMFRlc3QyJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHRlc3QoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBjb25zb2xlLmxvZygndGVzdDInKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlNUNuJTdEJTVDbiU1Q25UZXN0Mi5wcm90b3R5cGUuaWQlMjAlM0QlMjAnQSclM0IlNUNuJTVDbmV4cG9ydCUyMGNsYXNzJTIwVGVzdDMlMjAlN0IlNUNuJTIwJTIwJTIwJTIwdGVzdCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKCd0ZXN0MycpJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCU1Q24lN0QlNUNuJTVDblRlc3QzLmElMjAlM0QlMjAlN0IlN0QlM0IlNUNuVGVzdDMuYS5pZCUyMCUzRCUyMCdBJyUzQiU1Q24lNUNuJTVDbmV4cG9ydCUyMGNsYXNzJTIwVGVzdDQlMjAlN0IlNUNuJTIwJTIwJTIwJTIwdGVzdCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKCd0ZXN0NCcpJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCU1Q24lN0QlNUNuJTVDbk9iamVjdC5kZWZpbmVQcm9wZXJ0eShUZXN0NCUyQyUyMCdpZCclMkMlMjAlN0J2YWx1ZSUzQSUyMCdBJyU3RCklM0IlNUNuJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIwdGVzdCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKCd0ZXN0RycpJTNCJTVDbiU3RCU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==)

(original repl was using rollup 2.10.8)

Reply:
Next version of Terser will handle class property assignments as pure, so marking this as resolved.